entrypoint.py

```diff
@@ -233,23 +239,62 @@
 git = git.Repo().git
 git.config('--global', 'user.name', os.environ['INPUT_GIT_USER_NAME'])
 git.config('--global', 'user.email', os.environ['INPUT_GIT_USER_EMAIL'])
-token  = os.environ['RELEASE_TOKEN']
-remote = 'https://' + token + '@github.com/' + os.environ['GITHUB_REPOSITORY']
+if deploy_key:
+    @contextmanager
+    def git_auth():
+        """
+        Temporarily set SSH credentials for Git. To be used as context manager.
+        """
+        (ssh_wrapper_fd, ssh_wrapper_path) = mkstemp(text=True)
+        try:
+            with NamedTemporaryFile() as identity_file:
+                with open(ssh_wrapper_fd, "w") as ssh_wrapper_file:
+                    ssh_wrapper_file.write('#!/bin/sh\n')
+                    ssh_wrapper_file.write(
+                        f'exec ssh -o StrictHostKeyChecking=no '
+                        f'-i {identity_file.name} $@')
+                    os.chmod(ssh_wrapper_path, 0o500)
+
+                identity_file.write(deploy_key.encode('utf-8'))
+                identity_file.flush()
+                os.environ['GIT_SSH'] = ssh_wrapper_path
+                try:
+                    yield
+                finally:
+                    del os.environ['GIT_SSH']
+        finally:
+            os.unlink(ssh_wrapper_path)
+
+    remote = f'git@github.com:{os.environ["GITHUB_REPOSITORY"]}'
+else:
+    @contextmanager
+    def git_auth():
+        """
+        No-op context manager.
+        """
+        yield
+
+    token  = os.environ['RELEASE_TOKEN']
+    remote = f'https://{token}@github.com/{os.environ["GITHUB_REPOSITORY"]}'
 git.remote('add', 'gh-token', remote)
-git.fetch('gh-token')
-# reset will fail if 'generated-documentation` branch doesn't yet exist. That's
-# fine, it will exist after our push. Just not the error and move on.
-try:
-    git.reset('gh-token/generated-documentation')
-except GitCommandError:
-    print(NOTICE + "Couldn't git reset generated-documentation." + ENDC)
-    print(NOTICE + "This error is expected if the branch doesn't exist yet." \
-      + ENDC)
-
-print(INFO + "Running 'mkdocs gh-deploy'." + ENDC)
-os.chdir(docs_build_dir)
-rslt = os.system('mkdocs gh-deploy --verbose --clean \
-    --remote-name gh-token --remote-branch generated-documentation')
-if rslt != 0:
+with git_auth():
+    git.fetch('gh-token')
+    # reset will fail if 'generated-documentation` branch doesn't yet exist.
+    # That's fine, it will exist after our push. Just not the error and move on.
+    try:
+        git.reset('gh-token/generated-documentation')
+    except GitCommandError:
+        print(NOTICE + "Couldn't git reset generated-documentation." + ENDC)
+        print(NOTICE + "This error is expected if the branch doesn't exist yet."
+          + ENDC)
+
+    print(INFO + "Running 'mkdocs gh-deploy'." + ENDC)
+    # pylint: disable=W1510
```

Comment:
is this related to the `cwd` change?

what made this necessary? 

I think that an explicit: os.chdir(docs_build_dir) is more clear. and more likely to give a good error message, but I'm not sure about that.

Reply:
It suppresses the following pylint warning:

```
entrypoint.py:286:11: W1510: Using subprocess.run without explicitly set `check` is not recommended. (subprocess-run-check)
```

`check=True` will raise an exception if the process exits with status != 0. As we always check the status code, that's not an issue. Plus I think an exception in the output would be way more confusing than the error message we print instead.

As I'm not using any of subprocess's features anyway, I can also change the code back to `os.chdir` and `os.system`. I initially wanted to pass a modified environment to the subprocess, hence the change to the subprocess module.

Reply:
You know Python better than I, that's clear. Whatever you think is the easiest to maintain and for someone like me to understand, let's do that.